File: 백준/Silver/3273. 두 수의 합/두 수의 합.kt
package org.example

import java.util.StringTokenizer


fun main() {
    val br = System.`in`.bufferedReader()

    // 입력
    val n = br.readLine().toInt()
    val numbers = IntArray(n)
    val st = StringTokenizer(br.readLine())
    for (i in 0 until n) {
        numbers[i] = st.nextToken().toInt()
    }
    val x = br.readLine().toInt()
    // 

    val seen = mutableSetOf<Int>() // 등장한 숫자 저장
    var count = 0

    for (num in numbers) {
        if ((x - num) in seen) {
            // x에서 num 뺀 것이 존재한다면 쌍이 존재하는 것 (num, x-num)
            count++
        }
        seen.add(num) // 현재 숫자를 set에 추가
    }
    
    println(count)
    br.close()
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py

import sys

input = sys.stdin.readline

# 입력 처리
n = int(input())
arr = list(map(int, input().split()))
x = int(input())

##
seen = set()
count = 0

for num in arr:
    if x - num in seen:
        count += 1
    seen.add(num)

print(count)


